core/src/test/java/zipkin2/finagle/ZipkinTracerTest.java

```diff
@@ -53,21 +53,58 @@
   InMemoryStatsReceiver stats = new InMemoryStatsReceiver();
   BlockingQueue<List<Span>> spansSent = new LinkedBlockingDeque<>();
 
+  ZipkinTracer.Config config = new ZipkinTracer.Config() {
+    @Override public String localServiceName() {
+      return "unknown";
+    }
+
+    @Override public float initialSampleRate() {
+      return 1.0f;
+    }
+  };
+
   ZipkinTracer tracer = newTracer(FakeSender.create().onSpans(spansSent::add));
 
   ZipkinTracer newTracer(Sender sender) {
     return new ZipkinTracer(AsyncReporter.builder(sender)
         .messageTimeout(0, TimeUnit.MILLISECONDS)
         .messageMaxBytes(176 + 5) // size of a simple span w/ 128-bit trace ID + list overhead
         .metrics(new ReporterMetricsAdapter(stats))
-        .build(), () -> 1.0f, stats);
+        .build(), config, stats);
   }
 
   @After
-  public void closeTracer() throws Exception {
+  public void closeTracer() {
     tracer.close();
   }
 
+  @Test public void defaultIsLocalServiceNameFromAnnotation() throws Exception {
+    recordClientSpan(root);
+
+    assertThat(spansSent.take().stream())
+        .flatExtracting(Span::localServiceName)
+        .containsExactly("web");
+  }
+
+  @Test public void configOverridesLocalServiceName() throws Exception {
+    config = new ZipkinTracer.Config() {
+      @Override public String localServiceName() {
+        return "favstar";
+      }
+
+      @Override public float initialSampleRate() {
+        return 1.0f;
+      }
+    };
+    tracer = newTracer(FakeSender.create().onSpans(spansSent::add));
+
+    recordClientSpan(root);
+
+    assertThat(spansSent.take().stream())
+        .flatExtracting(Span::localServiceName)
+        .containsExactly("favstar");
```

Comment:
Is it appropriate here to set `remoteServiceName` to the value we are overwriting, in this case `web`?

Reply:
It is a heuristic we could use.. In the case that the serviceName we are overriding was actually just a bad local service name it wouldn't help. OTOH zipkin treats the server's service name authoritative..

Reply:
looped the idea back into the zipkin-user thread